lib/authentication/service/auth_provider.dart

```diff
@@ -435,4 +418,36 @@ class AuthProvider with ChangeNotifier {
     }
     return true;
   }
+
+  /// Update the user information with the data in [info].
+  Future<bool> updateProfile(
+      {Map<String, dynamic> info, BuildContext context}) async {
+    try {
+      String firstName = info[S.of(context).labelFirstName];
+      String lastName = info[S.of(context).labelLastName];
+
+      List<String> classes = info['class'] ?? null;
+
+      User user =
+          await Provider.of<AuthProvider>(context, listen: false).currentUser;
+      user.firstName = firstName;
+      user.lastName = lastName;
+      user.classes = classes;
+
+      Firestore.instance
+          .collection('users')
+          .document(user.uid)
+          .updateData(user.toData());
+
+      var userUpdateInfo = UserUpdateInfo();
+
```

Comment:
I meant add a newline before the `notifyListeners()` line, to separate different sections of the code. There are three lines that deal with updating the FirebaseUser info, and they should be together.